src/utils/history.js

```diff
@@ -196,3 +202,87 @@ export const getTokenTransactionsFromHistory = (
       (tag === PAYMENT_NETWORK_TX_SETTLEMENT && extra.find(({ symbol }) => symbol === token)),
   );
 };
+
+export const parseHistoryEventFee = (
+  feeWithGasToken: ?FeeWithGasToken,
+  gasUsed: ?number,
+  gasPrice: ?number,
+): ?TokenValue => {
+  if (feeWithGasToken?.feeInWei) {
+    const {
+      feeInWei,
+      gasToken: { decimals, symbol },
+    } = feeWithGasToken;
+    return {
+      value: wrapBigNumber(formatUnits(feeInWei, decimals)),
+      symbol,
+    };
+  }
+
+  if (gasUsed && gasPrice) {
+    const feeValue = wrapBigNumber(gasPrice).multipliedBy(wrapBigNumber(gasPrice));
```

Comment:
BN operations arguments seem to work fine with `number | string` too
```suggestion
    const feeValue = wrapBigNumber(gasPrice).multipliedBy(gasPrice);
```